lua/terrortown/menus/score/info.lua

```diff
@@ -3,128 +3,62 @@
-
-			filteredPlyScoreList[scoreName] = filteredPlyScoreList[scoreName] or {score = {0,0}, numEvents = {0,0}}
-
-			local posIndex = score < 0 and 1 or 2 -- first entry contains negative and second positive entries
-			local filteredPlyScore = filteredPlyScoreList[scoreName]
-
-			filteredPlyScore.score[posIndex] = filteredPlyScore.score[posIndex] + score
-			filteredPlyScore.numEvents[posIndex] = filteredPlyScore.numEvents[posIndex] + 1
+	local lengthLongestLine, titleHeight = GetPanelTextSize(titleBox)
+	local height = math.max(initTitleHeight, titleHeight)
+	titleBox:SetHeight(height)
+
+	for i = 1, #items do
+		local thing = items[i]
+		local plyRow = boxLayout:Add("DColoredTextBoxTTT2")
+		plyRow:SetDynamicColor(parent, 0)
+		plyRow:SetTitleAlign(TEXT_ALIGN_LEFT)
+		if thing.title then
+			plyRow:SetTitle(thing.title)
 		end
-	end
 
-	-- In a second pass we create the tooltip boxes with summarized scoreevents
-	for rawScoreTextName, scoreObj in pairs(filteredPlyScoreList) do
-		for i = 1, 2 do
-			local score = scoreObj.score[i]
-			local numberEvents = scoreObj.numEvents[i]
+		local rowWidth, rowHeight = GetPanelTextSize(plyRow)
 
-			if score == 0 then continue end
-
-			local plyRoleBox = boxLayout:Add("DColoredTextBoxTTT2")
-			plyRoleBox:SetSize(width, 20)
-			plyRoleBox:SetDynamicColor(parent, 0)
-			plyRoleBox:SetTitleAlign(TEXT_ALIGN_LEFT)
-
-			plyRoleBox.GetTitle = function()
-				return "- " .. (numberEvents > 1 and (numberEvents .. "x ") or "") .. ParT("tooltip_" .. rawScoreTextName, {score = score})
-			end
-
-			height = height + 20
+		if thing.iconMaterial then
+			plyRow:SetIcon(thing.iconMaterial)
+			rowWidth = rowWidth + iconOffset
 		end
-	end
 
-	return height
-end
-
-local function MakePlayerKarmaTooltip(parent, width, ply)
-	local plyKarmaList = CLSCORE.eventsPlayerKarma[ply.sid64] or {}
-	local height = 25
-
-	local boxLayout = vgui.Create("DIconLayout", parent)
-	boxLayout:Dock(FILL)
-
-	local titleBox = boxLayout:Add("DColoredTextBoxTTT2")
-	titleBox:SetSize(width, 25)
-	titleBox:SetDynamicColor(parent, 0)
-	titleBox:SetTitle("tooltip_karma_gained")
-	titleBox:SetTitleAlign(TEXT_ALIGN_LEFT)
+		if rowWidth > lengthLongestLine then
+			lengthLongestLine = rowWidth
+		end
 
-	if not istable(plyKarmaList) then
-		return height
+		rowHeight = math.max(lineHeight, rowHeight)
+		plyRow:SetHeight(rowHeight)
+		height = height + rowHeight
 	end
 
-	for karmaText, karma in pairs(plyKarmaList) do
-		local plyRoleBox = boxLayout:Add("DColoredTextBoxTTT2")
-		plyRoleBox:SetSize(width, 20)
-		plyRoleBox:SetDynamicColor(parent, 0)
-		plyRoleBox:SetTitleAlign(TEXT_ALIGN_LEFT)
+	-- add some padding to the right margin
+	lengthLongestLine = lengthLongestLine + iconOffset
 
-		plyRoleBox.GetTitle = function()
-			return "- " .. TryT(karmaText) .. ": " .. karma
-		end
-
-		height = height + 20
+	-- we found the longest line, now make all entries that long
+	for _, v in ipairs( boxLayout:GetChildren() ) do
```

Comment:
please do not use ipairs, `for i = 1, #xxx do` is much faster

Reply:
idk if it is still the case with the GetChild / ChildCount overhead but i never benched it

Reply:
you could do:

```lua
local children = boxLayout:GetChildren()

for i = 1, #children do
    local child = children[i]

end
```

This is how we do it based on this: https://springrts.com/wiki/Lua_Performance

Reply:
![image](https://github.com/TTT-2/TTT2/assets/5711436/a68b7cf5-6e40-4683-8c1b-d37a582e0b9c)
unless i bricked the measure of this script, including # in for structure measures worse in glua than it does in 5.1 that that site measures (it doesn't mention luajit anywhere either, which has mildly better perf of ipairs in general)

Reply:
Please use @TimGoll's approach, as it is close to our code style. 
Also in your test, ipairs seems slower, so i don't get what you are trying to say in your comment. Even though this is really niche and not noticable anyways.

(also regarding your latest commits, please use it the way Tim described above, because extra function calls in the loop to GetChild is more of a performance overhead)

Reply:
![image](https://github.com/TTT-2/TTT2/assets/5711436/4d9e607c-4f9d-4477-af1e-94f4ec4e147d)
ipairs is faster across longer consecutive iterations is what i mean, the performance profile is different. idk that we ever have a 10000 item long table, but it is still a characteristic difference